### DISCUSSION TITLE
NEXT BUILD MALFUNCTIONING ON VERCEL

### DISCUSSION THREAD

**Discussion by GeniusHawlah** [vercel/next.js#35589] [Help]
I built a blog with NextJS (geniushawlah.xyz) and realized that all other needed pages got indexed by Google crawlers except some pages, specifically my dynamic pages.
I checked the Output folders under page source on Vercel and realized the pages weren't generated truly. Which means, Google crawlers can't see while crawling. 

I reached out to Vercel days ago, after days of checking, the attendant assigned to me said the problem is "Next Specific" then gave me link to ask here.

**Comment by icyJoseph** [comment]
Are you using `next export`? `next build` + `next start`? 

How's your site setup? We need more information about the site, do you have a repository to take a look?

Are you using `getStaticPaths` with `getStaticProps`? Did you forget to use `fallback: true` on `getStaticPaths`?

**Reply by GeniusHawlah** [reply]
Nothing is wrong locally, the files are there in  .next folder. 
The thing is, Vercel engine builds it itself from my github commits. 

I'm using `getStaticPaths()` with `getStaticProps()` and using `fallback` as false since they are just few pages and I want to build all at once.

**Reply by icyJoseph** [reply]
Hi,

Yes, that's how Vercel deploys work. I think we may be making assumptions here, let's boil things down to the basics.

- Ok, so all pages are rendered at build time
- You have connected your next.js repo to Vercel and have fixed the Vercel scripts at `https://vercel.com/<username>/<project-name>/settings`

> Build & Development Settings
>
>When using a framework for a new project, it will be automatically detected. As a result, several project settings are automatically configured to achieve the best result. You can override them below.

It should be set to Next.js

- Are all files needed for the build, available in the repo?
- When you deploy with Vercel, you can see the logs produced per deployment under the `Deployment Overview > Building` - There you should see a summary of the output:

```
Page                                                 Size     First Load JS
--
┌ ● / (ISR: 10 Seconds) (2091 ms)                    14.1 kB         133 kB
├   /_app                                            0 B             101 kB
```

Something like that, does it contain all of the pages you expect? Are there any logs?

**Reply by GeniusHawlah** [reply]
All the files are available in the building log but not available when I check via the source tab


I replied via email also with the screenshots attached ![Screenshot_20220325-135600~2.png](https://user-images.githubusercontent.com/58650718/160125530-6be51aba-2ea6-454b-b2a0-872842973708.png)
![Screenshot_20220325-135632~2.png](https://user-images.githubusercontent.com/58650718/160125575-fdb03ff3-3038-460e-b1af-a32782b6be9c.png)

**Reply by icyJoseph** [reply]
Well, this is really strange. I just did a few teaks to a project of mine and re-deployed, and it does have things under blog, and such.

<img width="358" alt="Screenshot 2022-03-25 at 18 17 24" src="https://user-images.githubusercontent.com/21013447/160169759-b59f7565-f904-4b0d-90bb-e4b66bff440b.png">

From what I see in your build, you have the `●` symbol on blog/posts, which means these have been pre-generated.

Do you have any special configuration, or custom server for your Next.js project?

What Node.js version is your Vercel project using? 

I wonder if somehow `blog/posts` is overriding `blog/posts/[slug]`, but this seems very farfetched. 

And again, is it possible to take a look at this repository? At least the pages structure and the `next.config.js` file.

I understand your frustration, but I think we'll get to the bottom of this.

**Reply by GeniusHawlah** [reply]
- Yes the files were pre-generated. Both `blog/posts` and `blog/posts/[slug]`. And I don't think there is any technical reason for the former to be overriding the latter - truly farfetched. 

- No, I don't have custom config set for my build either locally or on Vercel setup page. 

- I'm currently on Node version 16.13.1 locally, while the Vercel build setting has version 14.x and that's the highest option there. 

- Lastly, the repo contains some private stuff, but I can assure you that nothing is wrong with the pages structure, and the next.config file h is pretty empty too.

**Comment by GeniusHawlah** [comment]
All the files are available in the building log but not available when I
check via the source tab
